any_parser/any_parser.py

```diff
@@ -258,15 +340,53 @@ def async_fetch(
         if response is None:
             return "Error: timeout, no response received"
         if response.status_code == 200:
-            markdown_list = response.json()["markdown"]
-            return "\n".join(markdown_list)
+            if result_type == "json":
+                return response.json()["json"]
+            else:
+                markdown_list = response.json()["markdown"]
+                return "\n".join(markdown_list)
         if response.status_code == 202:
             return None
         return f"Error: {response.status_code} {response.text}"
 
+    def _upload_file_to_presigned_url(
+        self, file_path: str, response: requests.Response
+    ) -> str:
+        if response.status_code == 200:
+            try:
+                file_id = response.json().get("fileId")
+                presigned_url = response.json().get("presignedUrl")
+                with open(file_path, "rb") as file_to_upload:
+                    files = {"file": (file_path, file_to_upload)}
+                    upload_resp = requests.post(
+                        presigned_url["url"],
+                        data=presigned_url["fields"],
+                        files=files,
+                        timeout=TIMEOUT,
+                    )
+                    if upload_resp.status_code != 204:
+                        return f"Error: {upload_resp.status_code} {upload_resp.text}"
+                return file_id
+            except json.JSONDecodeError:
+                return "Error: Invalid JSON response"
+        else:
+            return f"Error: {response.status_code} {response.text}"
+
     def _check_model(self, model: ModelType) -> None:
-        if model not in {ModelType.BASE, ModelType.PRO, ModelType.ULTRA}:
+        if model not in {ModelType.BASE, ModelType.PRO}:
             valid_models = ", ".join(["`" + model.value + "`" for model in ModelType])
-            raise ValueError(
-                f"Invalid model type: {model}. Supported `model` types include {valid_models}."
-            )
+            return f"Invalid model type: {model}. Supported `model` types include {valid_models}."
+
+    def _check_file_type_and_path(self, file_path, file_extension):
+        # Check if the file exists
+        if not Path(file_path).is_file():
+            return f"Error: File does not exist: {file_path}"
+
+        if file_extension not in SUPPORTED_FILE_EXTENSIONS:
+            supported_types = ", ".join(SUPPORTED_FILE_EXTENSIONS)
+            return f"Error: Unsupported file type: {file_extension}. Supported file types include {supported_types}."
+
+    def _check_result_type(self, result_type: str) -> None:
+        if result_type not in RESULT_TYPES:
+            valid_result_types = ", ".join(RESULT_TYPES)
+            return f"Invalid result type: {result_type}. Supported `result_type` types include {valid_result_types}."
```

Comment:
Refactor to a utils.py to improve readability and no need to use _ then.

Reply:
moved to utils.py in latest commit